packages/autocomplete/src/index.spec.ts

```diff
@@ -294,7 +294,7 @@ describe('completer.completer', function () {
 
     it('returns all suggestions', async function () {
       const i = 'db.';
-      const attr = shellSignatures.Database.attributes as any;
+      const attr = shellSignatures.DatabaseImpl.attributes as any;
```

Comment:
The fact that Database and Collection are now types and the classes got replaced with DatabaseImpl and CollectionImpl is having a lot of knock-on effects and kinda ended up being responsible for most of this diff.

I'm wondering if we should keep them as Database and Collection but then rather name the new types to be something like DatabaseWithSchema and CollectionWithSchema. Then maybe the change will be smaller?

Not sure which way around would work out better ergonomically overall.